pkg/bgpv1/test/neighbor_test.go

```diff
@@ -137,3 +139,142 @@ func Test_NeighborAddDel(t *testing.T) {
 		})
 	}
 }
+
+// Test_NeighborGracefulRestart tests graceful restart configuration knobs with single peer.
+func Test_NeighborGracefulRestart(t *testing.T) {
+	testutils.PrivilegedTest(t)
+
+	var steps = []struct {
+		description       string
+		neighbor          cilium_api_v2alpha1.CiliumBGPNeighbor
+		waitState         []string
+		expectedPeerState peeringState
+	}{
+		{
+			description: "add neighbor with defaults",
+			neighbor: cilium_api_v2alpha1.CiliumBGPNeighbor{
+				PeerAddress: dummies[instance1Link].ipv4.String(),
+				PeerASN:     int(gobgpASN),
+			},
+			waitState: []string{"ESTABLISHED"},
+			expectedPeerState: peeringState{
+				peerASN:     gobgpASN,
+				peerAddr:    dummies[instance1Link].ipv4.Addr().String(),
+				peerSession: types.SessionEstablished.String(),
+			},
+		},
+		{
+			description: "update graceful restart with defaults",
+			neighbor: cilium_api_v2alpha1.CiliumBGPNeighbor{
+				PeerAddress: dummies[instance1Link].ipv4.String(),
+				PeerASN:     int(gobgpASN),
+				GracefulRestart: cilium_api_v2alpha1.CiliumBGPNeighborGracefulRestart{
+					Enabled: true,
+				},
+			},
+			waitState: []string{"ESTABLISHED"},
+			expectedPeerState: peeringState{
+				peerASN:                gobgpASN,
+				peerAddr:               dummies[instance1Link].ipv4.Addr().String(),
+				peerSession:            types.SessionEstablished.String(),
+				gracefulRestartEnabled: true,
+				gracefulRestartTime:    int64(types.DefaultGRRestartTime.Seconds()),
+			},
+		},
+		{
+			description: "update graceful restart, restart time",
+			neighbor: cilium_api_v2alpha1.CiliumBGPNeighbor{
+				PeerAddress: dummies[instance1Link].ipv4.String(),
+				PeerASN:     int(gobgpASN),
+				GracefulRestart: cilium_api_v2alpha1.CiliumBGPNeighborGracefulRestart{
+					Enabled:     true,
+					RestartTime: meta_v1.Duration{Duration: 20 * time.Second},
+				},
+			},
+			waitState: []string{"ESTABLISHED"},
+			expectedPeerState: peeringState{
+				peerASN:                gobgpASN,
+				peerAddr:               dummies[instance1Link].ipv4.Addr().String(),
+				peerSession:            types.SessionEstablished.String(),
+				gracefulRestartEnabled: true,
+				gracefulRestartTime:    20,
+			},
+		},
+		{
+			description: "disable graceful restart",
+			neighbor: cilium_api_v2alpha1.CiliumBGPNeighbor{
+				PeerAddress: dummies[instance1Link].ipv4.String(),
+				PeerASN:     int(gobgpASN),
+				GracefulRestart: cilium_api_v2alpha1.CiliumBGPNeighborGracefulRestart{
+					Enabled: false,
+				},
+			},
+			waitState: []string{"ESTABLISHED"},
+			expectedPeerState: peeringState{
+				peerASN:                gobgpASN,
+				peerAddr:               dummies[instance1Link].ipv4.Addr().String(),
+				peerSession:            types.SessionEstablished.String(),
+				gracefulRestartEnabled: false,
+			},
+		},
+	}
+
+	// This test run can take upto a minute
+	testCtx, testDone := context.WithTimeout(context.Background(), 4*maxTestDuration)
```

Comment:
Nit: would be nice to have a separate const for this, same as we have for `Test_NeighborAddDel`: https://github.com/cilium/cilium/blob/50b964e65a124e8bac44e187e5e862d2f87b98b4/pkg/bgpv1/test/neighbor_test.go#L22-L26